Sources/_CertificateInternals/CMakeLists.txt

```diff
@@ -0,0 +1,22 @@
+##===----------------------------------------------------------------------===##
+##
+## This source file is part of the SwiftCertificates open source project
+##
+## Copyright (c) 2023 Apple Inc. and the SwiftCertificates project authors
+## Licensed under Apache License v2.0
+##
+## See LICENSE.txt for license information
+## See CONTRIBUTORS.txt for the list of SwiftCertificates project authors
+##
+## SPDX-License-Identifier: Apache-2.0
+##
+##===----------------------------------------------------------------------===##
+
+add_library(_CertificateInternals
+  "_TinyArray.swift")
+
+set_target_properties(_CertificateInternals PROPERTIES
+  INTERFACE_INCLUDE_DIRECTORIES ${CMAKE_Swift_MODULE_DIRECTORY})
+
+_install_target(_CertificateInternals)
+set_property(GLOBAL APPEND PROPERTY SWIFT_CERTIFICATES_EXPORTS _CertificateInternals)
```

Comment:
@yim-lee I'm adding a new internal target with this PR and have copied the make file from the X509 folder. I'm not familiar with cmake. Could you please have a look at this file and the changes in the other cmake related changes?

Reply:
I left a couple of comments regarding `Foundation` but otherwise cmake changes look good to me.

Reply:
Thank you Yim!